### ISSUE TITLE
Load more XWB files

### ISSUE THREAD

**Issue by qingsheng8848** [nipkownix/re4_tweaks#212]
### Describe the feature you'd like to see added

The original code only allows 6 XWB files about EMXX.UDAS。When we add 6 kinds of  enemies in a stage, the seventh enemy will mute.Key code at address BIO4.EXE + 56C1A6。We can use the code to load the enemy's file. Pushhad Push 00 Push 00 Push 24 (mean em24.udas) Call Bio4.exe + 844f Add ESP 0C POPAD RET。version 1.0.6。When the XWB file is successfully loaded. When the XWB file is successfully loaded. Will pass Bio4.exe + 575FCD

**Comment by qingsheng8848** [comment]
https://youtu.be/qYOGLzzyj3A
in this video.  at 1:29.  When playing shoot em3c, no sound。

**Comment by qingsheng8848** [comment]
![无标题](https://user-images.githubusercontent.com/98869895/173383866-59711eb6-5ac9-4fd6-8bf7-9fcbcf746a33.png)

**Comment by qingsheng8848** [comment]
![Uploading 1.png…]()

**Comment by qingsheng8848** [comment]
![标题](https://user-images.githubusercontent.com/98869895/173385122-892b4da4-3052-4727-adc6-6d33ec2aa9a6.png)

**Comment by qingsheng8848** [comment]
https://mega.nz/file/pxIgXb5D#P1ldP2gyNI-E3ZVR6f2SpTkWIiZknzcb531yl7H5gBM
test file。If the judgment command in the figure is modified to 07.  Then the cutscene is triggered after killing em12.  Then loading the enemy xwb file causes a crash

**Comment by emoose** [comment]
Seems the value checked against 6 is used as an offset into some byte array inside `pSnd`, I'd guess they probably only have room for 6 there, increasing it probably makes it overwrite data that isn't related to it.

If the bytes were at the end of `pSnd`, you could probably just increase size of buffer they use, but looks like the bytes are somewhere in the middle, would take some work to resize it since all the code that uses bytes after it would need to be changed, seems there's at least ~110 functions making use of `pSnd` directly, probably more that use it indirectly too :/

**Comment by qingsheng8848** [comment]
@emoose Yes, starting at address 0x1220234.  Each em.udas file loaded will take up four bytes. This table is in the middle.  Only 0x18 bytes can be used.  this is really a problem

**Comment by qingsheng8848** [comment]
So what is the reason for causing the game crash? Is the buffer not enough?

**Comment by emoose** [comment]
I guess it's writing past the 6 entries / 0x18 bytes and ends up overwriting something unrelated, like a pointer to something maybe.

If you can find all the code which accesses those 0x18 bytes maybe you could redirect it to your own buffer instead, might take some effort to find everything that uses it though... I also tried mapping out some of the `Snd` struct and looks like there's a bunch of other different 6-entry arrays which might be related to it as well, might also need those changed too :/

**Comment by qingsheng8848** [comment]
@emoose ，Ah, I will try, no matter what, thank you for your answer.
